botocore/session.py

```diff
@@ -472,6 +472,9 @@ def user_agent(self):
                                           platform.release())
         if self.user_agent_extra:
             base += ' %s' % self.user_agent_extra
+        if os.environ.get('AWS_EXECUTION_ENV') is not None:
+            base += ' exec-env/%s' % os.environ.get('AWS_EXECUTION_ENV')
```

Comment:
Also it would be good to add this to the doc string about the ``user_agent`` as all of the other components are documented..